app/services/authorizer.rb

```
@@ -104,11 +104,18 @@ def build_filtered_scope_components(resource_class, all_filters, options)
     return result if all_filters.any?(&:unlimited?)
 
     search_string = build_scoped_search_condition(all_filters.select(&:limited?))
-    find_options = ScopedSearch::QueryBuilder.build_query(resource_class.scoped_search_definition, search_string, options)
-    result[:where] << find_options[:conditions]
-    result[:includes].push(*find_options[:include])
-    result[:joins].push(*find_options[:joins])
-    result
+
+    begin
+      find_options = ScopedSearch::QueryBuilder.build_query(resource_class.scoped_search_definition, search_string, options)
+
+      result[:where] << find_options[:conditions]
+      result[:includes].push(*find_options[:include])
+      result[:joins].push(*find_options[:joins])
+
+      result
+    rescue ScopedSearch::QueryNotSupported => e
+      raise Foreman::InvalidScopeFilter, "#{resource_class}: #{e.message}"
```

I'm still on the fence here: do we want to throw an exception, or do we want to return a `none` scope in case the query is not correct?

---

We could log the error and return an empty scope, that's true. Will take a look at it.